src/ray/raylet/agent_manager.cc

```diff
@@ -132,21 +132,17 @@ void AgentManager::StartAgent() {
                   << exit_code << ". ip " << reported_agent_ip_address_ << ". id "
                   << reported_agent_id_;
 
-    if (exit_code == 0) {
-      RAY_LOG(INFO) << "Agent is terminated gracefully (exit code 0). "
-                    << "Raylet will shutdown gracefully.";
-      // Sending a SIGTERM to itself is equivalent to gracefully shutting down raylet.
-      RAY_CHECK(std::raise(SIGTERM) == 0) << "There was a failure while sending a "
-                                             "sigterm to itself. The process will not "
-                                             "gracefully shutdown.";
-    } else {
-      RAY_LOG(ERROR)
-          << "The raylet exited immediately because the Ray agent failed. "
-             "The raylet fate shares with the agent. This can happen because the "
-             "Ray agent was unexpectedly killed or failed. See "
-             "`dashboard_agent.log` for the root cause.";
-      QuickExit();
-    }
+    RAY_LOG(ERROR)
+        << "The raylet exited immediately because the Ray agent failed. "
```

Comment:
Should we change the error message to reflect this graceful shutdown and then forceful kill at 10secs timeout?

Reply:
I will do it as a follow up since I'd like to merge it by a branch cut